sdks/java/io/hadoop/input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java

```diff
@@ -832,7 +731,7 @@ public void close() throws IOException {
       public Double getFractionConsumed() {
```

Comment:
In a couple of places in this method there is an if statement where we read from a value that could come from another thread (doneReading/recordReader==null/recordsReturned), and then we set the progressValue based on that. 

This is a potential race condition since you are reading a value and then setting a value based on that read without taking a lock. 

I think it would be fine if instead of setting progressValue here, you simply return'ed. This okay because you're just returning a snapshot in time, which is correct. 

This should not block this PR, but it needs to get fixed in the next PR.